core/processor/src/main/java/io/quarkus/annotation/processor/documentation/config/scanner/AbstractConfigListener.java

```diff
@@ -66,6 +68,16 @@ public void onResolvedEnum(TypeElement enumTypeElement) {
         configCollector.addResolvedEnum(enumDefinition);
     }
 
+    protected void validateRuntimeConfigOnDeploymentModules(ConfigPhase configPhase) {
+        if (configPhase.equals(ConfigPhase.RUN_TIME) || configPhase.equals(ConfigPhase.BUILD_AND_RUN_TIME_FIXED)) {
+            ExtensionModule.ExtensionModuleType type = config.getExtensionModule().type();
+            if (type.equals(ExtensionModule.ExtensionModuleType.DEPLOYMENT)) {
+                throw new IllegalStateException("Configuration classes with ConfigPhase.RUN_TIME or " +
+                        "ConfigPhase.BUILD_AND_RUNTIME_FIXED phases, must be on runtime module.");
```

Comment:
```suggestion
                        "ConfigPhase.BUILD_AND_RUNTIME_FIXED phases must be on runtime module.");
```

Reply:
Also I think we should log the offending `ConfigMapping` class name in the message